### PR TITLE
Feat(eos_cli_config_gen): Add support for MACsec fallback to unprotected traffic

### LINKED ISSUE #3696: Add support for MACsec fallback to unprotected traffic
### Enhancement summary

Theres a TOI around this feature: https://www.arista.com/en/support/toi/eos-4-21-1f/14084-macsec-fallback-to-unprotected-traffic

And tech-library also includes it in the macsec deployment guide:
https://tech-library.arista.com/data_center/macsec/deployment_guide/#lab-topology

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Standard macsec deployment between DCs, allow configuration of "traffic unprotected allow":

https://avd.arista.com/4.6/roles/eos_cli_config_gen/docs/input-variables.html#mac-security

```
mac security
    fips restrictions
    !
    profile macsec-profile
        cipher aes256-gcm-xpn 
        key 0abcd1 7 055B515F231E1A5D1854114709585D7F7E757E62617217574F53
        key 0abcd2 7 055B505F731C4F5B4E54134308595D7E72717E67307640524F5658 fallback
        mka session rekey-period 30
        **traffic unprotected allow** 
        l2-protocol lldp bypass unauthorized 
```

### Describe the solution you would like

Not sure if this will be necessary but in case customer wants to allow traffic unencrypted if MKA sessions fail.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

Add support for MACsec fallback to unprotected traffic.

## Related Issue(s)

Fixes #3696 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Adding `traffic_unprotected` to `mac_security` data-model-
```
mac_security:
       traffic_unprotected:

            # Allow/drop the transmit/receive of unprotected traffic.
            action: <str; "allow" | "drop"; required>

            # Allow transmit/receive of encrypted traffic using operational SAK and block otherwise.
            allow_active_sak: <bool>
```

## How to test
Molecule test and test on EOS.

### PR Checklist

- [x] Check the command ordering on EOS.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/mac_security.schema.yml [hunk 1]
```diff
@@ -95,3 +95,16 @@ keys:
                       valid_values:
                         - bypass
                         - "bypass unauthorized"
+            traffic_unprotected:
+              type: dict
+              keys:
+                allow:
```
Comment (Vibhu-gslab) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/mac_security.schema.yml]: `drop` key is missing

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/mac-security-eth-po-entropy.md [hunk 2]
```diff
@@ -199,10 +199,12 @@ mac security
       key 1234c 7 <removed> fallback
       mka key-server priority 100
       mka session rekey-period 30
+      traffic unprotected allow
       sci
       l2-protocol lldp bypass unauthorized
    profile A2
       key 1234b 7 <removed>
+      traffic unprotected allow active-sak
```
Comment (Vibhu-gslab) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/mac-security-eth-po-entropy.md:207]: table missing for `traffic unprotected`

**Review by Vibhu-gslab — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/mac-security-eth-po-entropy.md (hunk 2 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/mac-security-eth-po-entropy.md:207]: Added

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/mac_security.schema.yml (hunk 1 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/mac_security.schema.yml]: Updated the data-model schema.

**Review by Vibhu-gslab — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/mac-security.j2 [hunk 3]
```diff
@@ -28,13 +28,17 @@ FIPS restrictions enabled.
 
 ###### Settings
 
-| Cipher | Key-Server Priority | Rekey-Period | SCI |
-| ------ | ------------------- | ------------ | --- |
+| Cipher | Key-Server Priority | Rekey-Period | SCI | Traffic Unprotected |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/mac-security.j2]:
```suggestion
| Cipher | Key-Server Priority | Rekey-Period | SCI | Traffic Unprotected Fallback |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/mac-security.j2 [hunk 4]
```diff
@@ -34,6 +34,13 @@ mac security
 {%         if profile.mka.session.rekey_period is arista.avd.defined %}
       mka session rekey-period {{ profile.mka.session.rekey_period }}
 {%         endif %}
+{%         if profile.traffic_unprotected.action is arista.avd.defined %}
+{%             set traffic_unprotected_cli = "traffic unprotected " ~ profile.traffic_unprotected.action %}
+{%             if profile.traffic_unprotected.action is arista.avd.defined("allow") and profile.traffic_unprotected.allow_active_sak is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/mac-security.j2]:
```suggestion
{%             if profile.traffic_unprotected.action == "allow" and profile.traffic_unprotected.allow_active_sak is arista.avd.defined(true) %}
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
Minor comments. Otherwise looks good to me.

**Review by MaheshGSLAB — APPROVED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/mac-security-eth-po-entropy.md [hunk 5]
```diff
@@ -140,9 +140,9 @@ FIPS restrictions enabled.
 
 ###### Settings
 
-| Cipher | Key-Server Priority | Rekey-Period | SCI |
-| ------ | ------------------- | ------------ | --- |
-| aes128-gcm | 100 | 30 | True |
+| Cipher | Key-Server Priority | Rekey-Period | SCI | Traffic Unprotected Fallback |
+| ------ | ------------------- | ------------ | --- | ------------------- |
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/mac-security-eth-po-entropy.md:144]:
```suggestion
| ------ | ------------------- | ------------ | --- | ---------------------------- |
```

Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/mac-security-eth-po-entropy.md:144]:
```suggestion
| Cipher | Key-Server Priority | Rekey-Period | SCI | Traffic Unprotected Fallback |
| ------ | ------------------- | ------------ | --- | ---------------------------- |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/mac-security.j2 [hunk 6]
```diff
@@ -28,13 +28,17 @@ FIPS restrictions enabled.
 
 ###### Settings
 
-| Cipher | Key-Server Priority | Rekey-Period | SCI |
-| ------ | ------------------- | ------------ | --- |
+| Cipher | Key-Server Priority | Rekey-Period | SCI | Traffic Unprotected Fallback |
+| ------ | ------------------- | ------------ | --- | ------------------- |
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/mac-security.j2]:
```suggestion
| Cipher | Key-Server Priority | Rekey-Period | SCI | Traffic Unprotected Fallback |
| ------ | ------------------- | ------------ | --- | ---------------------------- |
```

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/mac-security.j2]: Fixed

**Review by MaheshGSLAB — APPROVED** [review]

**Review by gmuloc — APPROVED** [review]

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**